core/src/test/java/org/infinispan/container/offheap/OffHeapBoundedSingleNodeStressTest.java

```diff
@@ -1,30 +1,40 @@
 package org.infinispan.container.offheap;
 
-import org.infinispan.configuration.cache.CacheMode;
-import org.infinispan.configuration.cache.ConfigurationBuilder;
-import org.infinispan.configuration.cache.StorageType;
-import org.infinispan.eviction.EvictionType;
-import org.testng.annotations.Test;
-
 import java.util.Map;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorCompletionService;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+import org.infinispan.Cache;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.commons.marshall.WrappedBytes;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.infinispan.container.DataContainer;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.eviction.EvictionType;
+import org.testng.annotations.Test;
+
 /**
  * Test for native memory fragmentation when the values have different size.
  *
  * @author Dan Berindei
+ * @author William Burns
  */
 @Test(groups = "stress", testName = "container.offheap.OffHeapBoundedSingleNodeStressTest")
 public class OffHeapBoundedSingleNodeStressTest extends OffHeapMultiNodeStressTest {
 
    @Override protected void createCacheManagers() throws Throwable {
       ConfigurationBuilder dcc = getDefaultClusteredCacheConfig(CacheMode.LOCAL, false);
-      dcc.memory().storageType(StorageType.OFF_HEAP).evictionType(EvictionType.COUNT).size(2000);
+      dcc.memory().storageType(StorageType.OFF_HEAP).evictionType(EvictionType.COUNT)
+            .size(500)
+            // Set the address count to minimum to cause as many issues as possible!
+            .addressCount(1);
```

Comment:
I'd set it to 2, in the unlikely case an issue only appears when two buckets are modified concurrently.

Reply:
Technically it will never be 1 or even 2 normally, I just put a low number. This will essentially become `roundUpToPowerOf2(#CPU) *2`

Reply:
I forgot about that, I thought you could actually force the container to use a hash table with a single linked list :)